src/pages/BlockDetail/index.tsx

```diff
@@ -50,10 +51,18 @@ export default () => {
 
   const pageSize = queryBlockTransactions.data?.pageSize ?? pageSizeParam
 
+  const blockHeight = !isTransactionHash(blockHeightOrHash) ? blockHeightOrHash : queryBlock.data?.number
```

Comment:
Perhaps we should exclude string from the type of this value.

```suggestion
  const blockHeight = !isTransactionHash(blockHeightOrHash) ? Number(blockHeightOrHash) : queryBlock.data?.number
```

Reply:
> Perhaps we should exclude string from the type of this value.

Could be `NaN`